cli/stream_command.go

```diff
@@ -2326,22 +2326,36 @@ func (c *streamCmd) addAction(pc *fisk.ParseContext) (err error) {
 }
 
 func (c *streamCmd) rmAction(_ *fisk.ParseContext) (err error) {
-	c.connectAndAskStream()
-
 	if !c.force {
+		c.connectAndAskStream()
 		ok, err := askConfirmation(fmt.Sprintf("Really delete Stream %s", c.stream), false)
 		fisk.FatalIfError(err, "could not obtain confirmation")
 
 		if !ok {
 			return nil
 		}
-	}
 
-	stream, err := c.loadStream(c.stream)
-	fisk.FatalIfError(err, "could not remove Stream")
+		stream, err := c.loadStream(c.stream)
+		fisk.FatalIfError(err, "could not remove Stream")
+
+		err = stream.Delete()
+		fisk.FatalIfError(err, "could not remove Stream")
+	} else { // force delete attempt, even if somehow you can not get info on the stream in question (e.g. it's "offline")
+		c.nc, c.mgr, err = prepareHelper("", natsOpts()...)
+		fisk.FatalIfError(err, "setup failed")
+
+		var resp api.JSApiStreamDeleteResponse
 
-	err = stream.Delete()
-	fisk.FatalIfError(err, "could not remove Stream")
+		msg, err := c.nc.Request(fmt.Sprintf(api.JSApiStreamDeleteT, c.stream), []byte(""), time.Second)
```

Comment:
As per slack, I'd rather not do API calls direct here. Happy to handle this issue next time in the code but the consistent thing to do would be to add a method on the jsm package to do a delete by calling the API, I'd also do that call once the normal route fails

Reply:
We presumably also need this for consumers since the new server relocate code also leaves orphaned consumers when you try to use that